Tests/tests.py

```diff
@@ -1,118 +1,239 @@
-        app.db.session.close()
-        app.db.drop_all()
-        app.db.create_all()
-        self.client = app.flask_app.test_client()
+        flask_app.config.from_object('flask_settings.config_default.TestingConfig')
+        db.session.close()
+        db.drop_all()
+        db.create_all()
+        self.client = flask_app.test_client()
 
     def test_user_registration(self):
         """ Tests that checks if a user is able to register successfully"""
-        payload = {"first_name": "Sharon", "last_name": "Njihia",
-                   "email": "sharonkarendi5@gmail.com", "password": "password"}
-        response = self.client.post("/v1/auth/register", data=json.dumps(payload),
-                                    content_type='application/json')
-        self.assertEqual(response.status_code, 201)
+        with flask_app.app_context():
+            # create a user
+            payload = {"first_name": "Sharon", "last_name": "Njihia",
+                       "email": "sharonkaren@gmail.com", "password": "password"}
+            response = self.client.post(url_for('user_registration'), data=json.dumps(payload),
+                                        content_type='application/json')
+            self.assertEqual(response.status_code, 201)
 
     def test_the_user_payload_being_sent(self):
         """ Tests that the payload with user data has all the required info"""
-        payload = {}
-        response = self.client.post("/v1/auth/register", data=json.dumps(payload),
-                                    content_type="application/json")
-        self.assertEqual(response.status_code, 400)
+        with flask_app.app_context():
+            # create a user
+            payload = {}
+            response = self.client.post(url_for('user_registration'), data=json.dumps(payload),
+                                        content_type='application/json')
+            self.assertEqual(response.status_code, 400)
 
     def test_the_user_exists(self):
         """ Tests that when a user exists , they are not registered again"""
-        #  register the person
-        payload = {"first_name": "Sharon", "last_name": "Njihia",
-                   "email": "sharonkarendi5@gmail.com", "password": "password"}
-        response = self.client.post("/v1/auth/register", data=json.dumps(payload),
-                                    content_type='application/json')
-        self.assertEqual(response.status_code, 201)
-
-        # try and register the user again the status code should be 401
-        payload = {"first_name": "Sharon", "last_name": "Njihia",
-                   "email": "sharonkarendi5@gmail.com", "password": "password"}
-        response = self.client.post("/v1/auth/register", data=json.dumps(payload),
-                                    content_type='application/json')
-        self.assertEqual(response.status_code, 401)
+        with flask_app.app_context():
```

Comment:
If this class inherits the `BaseTestCase` you'd only need to check if the details used in `SetUp` can also be used to set up another user